### PR TITLE
Release v34 fixups

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Why is this needed?

This addresses feedback on v34. It also adds an additional LP bug for one of our fixes as I agreed to track in on the higher visibility bug in addition to the generic bug.

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

**Review by dheyay — APPROVED** [review]

File: debian/changelog [hunk 1]
```diff
@@ -44,7 +44,7 @@ ubuntu-advantage-tools (33) oracular; urgency=medium
     - contracts: add support for contracts which target a specific series
     - fips: change enable functionality to ensure all packages with a FIPS
       candidate are upgraded to the FIPS version (GH: #2667)
-    - fix:
+    - fix:
```
Comment (panlinux) [inline debian/changelog:47]: Since this is up for review... Extra whitespace at the end of the line? ;)

File: debian/changelog [hunk 2]
```diff
@@ -707,7 +707,7 @@ ubuntu-advantage-tools (27.11~22.10.1) kinetic; urgency=medium
     - pro:
       + Ubuntu Pro is released as a product
       + make `pro` the recommended executable for the client
-      + client, apt and motd messages updated/rewritten to show Pro
+      + client, apt and motd messages updated/rewritten to show Pro
```
Comment (panlinux) [inline debian/changelog:710]: Since this is up for review... Extra whitespace at the end of the line? ;)

File: apt-hook/20apt-esm-hook.conf [hunk 3]
```diff
@@ -3,5 +3,5 @@ APT::Update::Pre-Invoke {
 };
 
 binary::apt::AptCli::Hooks::Upgrade {
-	"[ ! -f /usr/lib/ubuntu-advantage/apt-esm-json-hook ] || /usr/lib/ubuntu-advantage/apt-esm-json-hook 2> /var/lib/ubuntu-advantage/apt-upgrade-hook-err.txt || true";
+	"[ ! -f /usr/lib/ubuntu-advantage/apt-esm-json-hook ] || /usr/lib/ubuntu-advantage/apt-esm-json-hook 2>> /var/log/ubuntu-advantage-apt-hook.log || true";
```
Comment (panlinux) [inline apt-hook/20apt-esm-hook.conf:6]: I checked and `debian/ubuntu-pro-client.logrotate` includes this new log file in its existing globbing rule.

File: debian/changelog (hunk 2 above)
Comment (orndorffgrant) [reply debian/changelog:710]: @panlinux this is in response to your comment here :) https://github.com/canonical/ubuntu-pro-client/pull/3247#discussion_r1739416521

Comment (panlinux) [reply debian/changelog:710]: Ah, you are undoing that change, ok, I should diff against pkg/ubuntu/devel

File: apt-hook/20apt-esm-hook.conf (hunk 3 above)
Comment (panlinux) [reply apt-hook/20apt-esm-hook.conf:6]: Did some testing, forcing a logrotate while the log file was empty, and later populating it with some garbage. It worked as expected. Here is when I forcibly populated the file, and then issue logrotate a few times:
```
$ ls -la /var/log/ubuntu-advantage*
-rw-r--r-- 1 root root    0 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log
-rw-r--r-- 1 root root    6 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log.1
-rw-r--r-- 1 root root   26 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log.2.gz
-rw-r--r-- 1 root root    0 Sep  4 18:42 /var/log/ubuntu-advantage.log
-rw-r--r-- 1 root root 2730 Sep  4 18:42 /var/log/ubuntu-advantage.log.1
-rw-r----- 1 root root 7737 Sep  4 18:38 /var/log/ubuntu-advantage.log.2.gz
```
And then the files are zero-sized, as it is in the snapshot above, logrotate correctly does nothing to them.

File: debian/changelog (hunk 2 above)
Comment (panlinux) [reply debian/changelog:710]: Did the diff against pkg/ubuntu/devel, no noise, just the new d/changelog entry, all good.

File: apt-hook/20apt-esm-hook.conf (hunk 3 above)
Comment (panlinux) [inline apt-hook/20apt-esm-hook.conf:6]: We need to include the removal of `/var/log/ubuntu-advantage-apt-hook.log*` in `debian/ubuntu-pro-client.postrm`

**Review by panlinux — CHANGES_REQUESTED** [review]
We need to include the removal of `/var/log/ubuntu-advantage-apt-hook.log*` in `debian/ubuntu-pro-client.postrm`

**Comment by orndorffgrant** [comment]
@panlinux I'm aware of that postinst snippet. It is old cruft in its current form. We may want to remove it and add something similar to ubuntu-pro-client.postinst, but that is not urgent and its current form is not harmful. I think it's okay to leave as is for this release.

**Comment by orndorffgrant** [comment]
> We need to include the removal of `/var/log/ubuntu-advantage-apt-hook.log*` in `debian/ubuntu-pro-client.postrm`

Added!

**Comment by panlinux** [comment]
rechecking...

**Review by panlinux — APPROVED** [review]
+1

**Pull request merged**